code/API_definitions/Get DeviceLocation.yaml

```diff
@@ -0,0 +1,237 @@
+openapi: 3.0.3
+info:
+  title: Get Device location API
+  description: Service Enabling Network Function API for location retrieval from a device identifier.
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.1.0-wip
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/
+security:
+  - oAuth2ClientCredentials: []
+#  - BasicAuth: []
+#  - apiKey: []
+  - three_legged:
+    - device-location-provide
+servers:
+  - url: '{apiRoot}/{basePath}'
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+      basePath:
+        default: location/v0
+        description: Base path for the device location API
+paths:
+  /locations:
+    get:
+      tags:
+        - Location retrieval
+      summary: 'Retrieve location for a user equipment'
+      operationId: retrieveLocation
+      description: Retrieve location for an user equipement from an identifier like phone number, IP address (and port) or external identifier provided by the network operator.
+      parameters:

```

Comment:
Here we have to decide whether we model this as GET or as POST. GET is more natural options to retrieve data but it has the problem of modelling complex input parameters. If we wanted to reuse the same `device` object that we have in QoD or the original spec of Location, we could pass an object as query parameter, which is valid, but it has to be encoded properly, which is less readable. Reusing the object as it is has the advantages that it is possible to express the constraints between the different properties, for example that port is required along with public IP address.
Modelling as POST is also possible and align with other APIs but guidelines are not very clear about this. We could model the endpoint as POST /retrieve-location, so options would be:

UrlEncoded( GET location/v0/locations?device={"ipV4Address":{"public_ip_address":"1.2.3.4","public_port":1234}})

or

POST /location/v0/retrieve-location
{
  "device": {
    "ipV4Address": {
      "public_ip_address": "1.2.3.4",
      "public_port": 1234
    }
  }
)

There are similar cases in other APIs so maybe it makes sense to raise this in commonalities and adjust guidelines accordingly

Reply:
Yes this is a point to be discussed. 
Looking in other API we have POST in device status right now and GET for Device-identifier

i will raise this point in commonalities but not sure we'll get a rule.

Regarding our case, I will probably prefer to use POST pattern to align with Check. 

Let's tab this point for next meeting.

Reply:
Agree that POST is probably a better way as the input is complex and it aligns with both device status as well as with verify location.